packages/patrol_cli/test/features/drive/test_runner_test.dart

```diff
@@ -230,7 +230,10 @@ void main() {
 
           final expectedLog = <String>[];
           expect(actualLog, equals(<String>[]));
-          unawaited(testRunner.run());
+          late TestRunnerResult result;
+          unawaited(() async {
+            result = await testRunner.run();
+          }());
```

Comment:
The other tests below use regular `await`s to get the results – much better since you won't run into the risk of the log output not catching up with the intermediate delays

Reply:
I'm doing it this way because of https://github.com/dart-lang/fake_async/issues/51.

BTW the delays are entirely controlled by the fake async, so if the log output "won't catch up", then it's a bug and it's good that tests will fail in that case.

Reply:
Something like expectLater won't work the way you want?

Reply:
If you have an idea how to make this work, I'll be happy to hear it.